prime-router/src/main/kotlin/tokens/AuthenticatedClaims.kt

```diff
@@ -1,135 +1,227 @@
-        // should return, resp.:  (ignore, true), <nothing>, (all-in-one-health-ca, true, (az_phd, false, (ignore, true)
-        memberships.forEach {
-            if (it == oktaSystemAdminGroup) return@forEach // skip
-            if (it.startsWith(oktaSenderGroupPrefix)) return Pair(it.removePrefix(oktaSenderGroupPrefix), true)
-            if (it.startsWith(oktaGroupPrefix)) return Pair(it.removePrefix(oktaGroupPrefix), false)
+    fun authorizedForSubmission(
+        requiredOrganization: String,
+        requiredSender: String? = null,
+        request: HttpRequestMessage<String?>
+    ): Boolean {
+        if (requiredOrganization.isBlank()) {
+            logger.warn(
+                "Unauthorized.  Missing required Org" +
+                    " for user ${this.userName}: ${request.httpMethod}:${request.uri.path}."
+            )
+            return false
+        }
+        // User must have one of these scopes to be authorized
+        val requiredScopes = mutableSetOf(
+            Scope.primeAdminScope,
+            "$requiredOrganization.*.user", // eg, md-phd.*.user
+            "$requiredOrganization.*.admin", // eg, md-phd.*.admin
+            "$requiredOrganization.*.report", // eg, md-phd.default.report
+        )
+        if (requiredSender != null) {
+            requiredScopes += "$requiredOrganization.$requiredSender.user" // eg, md-phd.default.user
+            requiredScopes += "$requiredOrganization.$requiredSender.admin" // eg, md-phd.default.admin
+            requiredScopes += "$requiredOrganization.$requiredSender.report" // eg, md-phd.default.report
+        }
+        return if (authorized(requiredScopes)) {
+            logger.info(
+                "Authorized request by user with claims ${this.scopes}" +
+                    " to submit data via client id $requiredOrganization.  Beginning to ingest report"
+            )
+            true
+        } else {
+            logger.warn(
+                "Invalid Authorization for user ${this.userName}: ${request.httpMethod}:${request.uri.path}." +
+                    " ERR: Claims from user are ${this.scopes} but required scopes are $requiredScopes"
+            )
+            false
         }
-        return null
     }
 
     /**
-     * Derive whether this user is an Admin based on claims [memberships].
-     * @return true if a well-formed Administrator claim is in [memberships].  False otherwise.
+     * @return true if these claims authorize access to the [requiredScopes].  False if unauthorized.
      */
-    private fun isPrimeAdmin(memberships: Collection<String>): Boolean {
-        memberships.forEach {
-            if (it == oktaSystemAdminGroup) return true
-        }
-        return false
+    fun authorized(requiredScopes: Set<String>): Boolean {
+        return Scope.authorized(this.scopes, requiredScopes)
     }
 
-    companion object {
+    companion object : Logging {
         /**
```

Comment:
This big swath of green code here is not really new, its copied over from AuthenticationStrategy.kt, which has been deleted.